src/Smalot/PdfParser/PDFObject.php

```diff
@@ -127,100 +132,195 @@ public function getContent(): ?string
         return $this->content;
     }
 
-    public function cleanContent(string $content, string $char = 'X')
+    public function cleanContent(?string $content): string
     {
-        $char = $char[0];
-        $content = str_replace(['\\\\', '\\)', '\\('], $char.$char, $content);
-
-        // Remove image bloc with binary content
-        preg_match_all('/\s(BI\s.*?(\sID\s).*?(\sEI))\s/s', $content, $matches, \PREG_OFFSET_CAPTURE);
-        foreach ($matches[0] as $part) {
-            $content = substr_replace($content, str_repeat($char, \strlen($part[0])), $part[1], \strlen($part[0]));
+        if (null == $content) {
+            return '';
         }
 
-        // Clean content in square brackets [.....]
-        preg_match_all('/\[((\(.*?\)|[0-9\.\-\s]*)*)\]/s', $content, $matches, \PREG_OFFSET_CAPTURE);
-        foreach ($matches[1] as $part) {
-            $content = substr_replace($content, str_repeat($char, \strlen($part[0])), $part[1], \strlen($part[0]));
+        // Find all strings () and replace them so they aren't affected
```

Comment:
What do you mean with strings here? These brackets? `()`

Reply:
In a document-stream context, anything delimited by parentheses are strings.

> The following are valid literal strings:
> 
>   (This is a string)
>   (Strings may contain newlines
>   and such.)
>   (Strings may contain balanced parentheses () and
>   special characters (*!&}^% and so on).)
>   (The following is an empty string.)
>   ()
>   (It has zero (0) length.)

- PDF Reference 1.7, Section 3.2.3 String Objects